spec/system/consumer/checkout/payment_spec.rb

```diff
@@ -58,6 +57,18 @@
       visit checkout_path
     end
 
+    context "when a line item is out of stock" do
+      before do
+        variant.on_demand = false
+        variant.on_hand = 0
+        variant.save!
+      end
+
+      it "returns me to the cart with an error message" do
+        out_of_stock_check(:payment)
+      end
+    end
```

Comment:
I noticed that you follow this context pattern quite strictly. It's seen as good style by many people. I personally find that it's good for larger tests but find it much easier to write short tests in a simple `it` block in this style:

```rb
it "alerts me about out of stock items" do
  clear_stock_for_variant
  visit step
  expect_out_of_stock_message
end
```

Some people follow this clear three part structure for tests: setup, action, expectation. Just FYI, you don't have to change anything.

Reply:
Ok, that makes sense as well. I understand having the setup within the context block is more code lines - I feel it made my reading easier, when set up in this way. But I totally see the redundancy, for more experienced people.

I'll try to keep this in mind for next time. Thanks again :pray: